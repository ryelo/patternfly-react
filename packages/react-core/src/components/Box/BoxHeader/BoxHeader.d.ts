import { HTMLProps } from 'react';

export interface BoxProps extends HTMLProps<HTMLDivElement> {
  children?: React.ReactNode;
}

declare const BoxHeader: React.SFC<BoxProps>;

export default BoxHeader;
