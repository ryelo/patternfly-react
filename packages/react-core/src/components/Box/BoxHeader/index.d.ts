export {
  default as BoxHeader,
  BoxProps
} from './BoxHeader';
