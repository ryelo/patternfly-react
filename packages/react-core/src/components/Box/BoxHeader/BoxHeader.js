import React from 'react';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';
import styles from '@patternfly/patternfly-next/components/Box/styles.css';

const propTypes = {
  /** content rendered inside the boxHeader */
  children: PropTypes.node,
  /** additional classes added to the boxHeader */
  className: PropTypes.string
};

const defaultProps = {
  children: '',
  className: ''
};

const BoxHeader = ({ children, className, ...props }) => (
  <div {...props} className={css(styles.boxHeader, className)}>
    {children}
  </div>
);

BoxHeader.propTypes = propTypes;
BoxHeader.defaultProps = defaultProps;

export default BoxHeader;
