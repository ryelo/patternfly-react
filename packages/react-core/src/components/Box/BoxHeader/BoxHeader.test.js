import BoxHeader from './BoxHeader';
import React from 'react';
import { shallow } from 'enzyme';

test('Box has Header', () => {
  const view = shallow(<BoxHeader>Box Footer</BoxHeader>);
  expect(view).toMatchSnapshot();
});
