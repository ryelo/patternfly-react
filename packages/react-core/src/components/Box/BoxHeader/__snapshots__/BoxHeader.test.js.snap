// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box has Header 1`] = `
.pf-c-box {
  display: flex;
  flex-direction: column;
  background-color: #ffffff;
  box-shadow: 0 0.0625rem 0.125rem 0 rgba(3, 3, 3, 0.2);
}

<div
  className="pf-c-box"
>
  Box Footer
</div>
`;
